src/commands/domain.js

```diff
@@ -89,4 +92,206 @@ async function rm (params) {
+  const expectedDnsForPublicLoadBalancer = await getDefaultLoadBalancersDnsInfo({ id: ownerId, appId }).then(sendToApi);
+
+  const loadBalancerDnsConfig = {
+    aRecords: expectedDnsForPublicLoadBalancer[0].dns.a,
+    cnameRecord: expectedDnsForPublicLoadBalancer[0].dns.cname,
+  };
+
+  const filteredDomains = app.vhosts.filter(({ fqdn }) => fqdn.includes(domain));
+  const domains = getParsedDomains(filteredDomains);
+  const sortedDomains = sortDomains(domains);
+
+  for (const { hostname, pathPrefix, isApex } of sortedDomains) {
+    const resolvedDnsConfig = {
+      aRecords: await resolver.resolveA(hostname),
+      cnameRecords: await resolver.resolveCname(hostname) ?? [],
+    };
+
+    const diagnosticResults = diagDomainConfig({ hostname, pathPrefix, isApex }, resolvedDnsConfig, loadBalancerDnsConfig);
+    allDomainDiagnostics.push({ ...diagnosticResults, resolvedDnsConfig });
+
+    if (diagnosticResults.diagSummary === 'invalid' || diagnosticResults.diagSummary === 'no-config') {
+      hasError = true;
+    }
+  }
+
+  switch (format) {
+    case 'json':
+      Logger.printJson(allDomainDiagnostics);
+      break;
+    default: {
+      if (allDomainDiagnostics.length > 0) {
+        Logger.println(colors.yellow('\n⚠ TODO avec David, une phrase pour dire qu\'on ne supporte pas tout + se référer à la doc de la commande + un lien vers feedback ?'));
+
+      }
+      allDomainDiagnostics.forEach((diag) => reportDomainDiagnostics(diag));
+
+      if (allDomainDiagnostics.length === 0 && !hasDomainFilter) {
+        Logger.println(`\nNo domain associated to the app "${app.name}" (${app.id})`);
+      }
+
+      if (allDomainDiagnostics.length === 0 && hasDomainFilter) {
+        Logger.println(`\nNo domain matches "${domain}" for the app "${app.name}" (${app.id})`);
+      }
+    }
+  }
+
+  if (hasError) {
+    throw new Error('At least one of the domains is misconfigured');
+  }
+}
+
+/** @param {DomainDiag & { resolvedDnsConfig: ResolveDnsResult }} domainDiag */
+function reportDomainDiagnostics ({ hostname, pathPrefix, resolvedDnsConfig, diagDetails, diagSummary }) {
+
+  const validDiags = diagDetails.filter((diag) => diag.code === 'valid-a');
+  const unknownDiags = diagDetails.filter((diag) => diag.code === 'unknown-a');
+  const missingDiags = diagDetails.filter((diag) => diag.code === 'missing-a');
+  const suggestedCname = diagDetails.find((diag) => diag.code === 'suggested-cname');
+  const missingCname = diagDetails.find((diag) => diag.code === 'missing-cname');
+  const unknownCname = diagDetails.find((diag) => diag.code === 'unknown-cname');
+
+  const hasARecords = resolvedDnsConfig.aRecords.length > 0;
+  const hasCnameRecord = resolvedDnsConfig.cnameRecords.length > 0;
+
+  Logger.println('\n' + colors.blue(hostname) + ' ' + colors.yellow(pathPrefix) + '\n');
+
+  switch (diagSummary) {
+    case 'managed':
+      Logger.println(' '.repeat(2) + colors.green('✔ Managed by Clever Cloud'));
```

Comment:
suggestion: extract `' '.repeat(2)` into a const `const indent2 = ' '.repeat(2);`, and use it everywhere below.